src/librustc/mir/visit.rs

```diff
@@ -675,20 +675,12 @@ macro_rules! make_mir_visitor {
                             place: & $($mutability)? Place<'tcx>,
                             context: PlaceContext,
                             location: Location) {
-                match place {
-                    Place::Base(place_base) => {
-                        self.visit_place_base(place_base, context, location);
-                    }
-                    Place::Projection(proj) => {
-                        let context = if context.is_mutating_use() {
-                            PlaceContext::MutatingUse(MutatingUseContext::Projection)
-                        } else {
-                            PlaceContext::NonMutatingUse(NonMutatingUseContext::Projection)
-                        };
+                let Place { base, projection } = place;
 
-                        self.visit_place(& $($mutability)? proj.base, context, location);
-                        self.visit_projection(proj, location);
-                    }
+                self.visit_place_base(base, context, location);
```

Comment:
the mutability errors happen because the `context` here has not been mutated through all the projections. I guess pass down the base as an argument to `visit_projection` and once you hit a `None`, call `visit_place_base`

Reply:
True, that was lost on the rebase. Unsure why do you want that to happen there. We can just add the `context` adjustments to `visit_place` the only problem is that implementors would need to do the same :/.

Reply:
what would be the alternative? wrap the `context` mutation in an `if projection.is_some()`?

Reply:
Yep as you can see https://github.com/rust-lang/rust/pull/60913/commits/2c82f6d817e6aa18a85624cb4bf694c1433c021f